todo_app/app.py

```diff
@@ -1,57 +1,159 @@
 
-#error handling for 404
+status_mapping = []
+board_Name = "ToDo"
+board_Id = ""
+
+def build_status_mapping():
+    trello_Board = TrelloBoard()
+    result = trello_Board.get_BoardList()
+    if result.status_code == 200:
+        for item in result.json():
+            if item['name'] == board_Name:
+                board_Id = item['id']   
+    else:
+        return render_template("404.html", error="Trello board 'ToDo' not found!")
+
+    if not (board_Id is None):
+        trello_List = TrelloList()
+        lists = trello_List.get_List(id=board_Id)
+        if lists.status_code == 200:
+            for list in lists.json():
+                if list['name'] == 'To Do':
+                    status_mapping.append ({
+                        'id' : list['id'],
+                        'status' : "Not Started"
+                    })
+                elif list['name'] == 'Doing':
+                    status_mapping.append ({
+                        'id' : list['id'],
+                        'status' : "In Progress"
+                    })
+                elif list['name'] == 'Done':
+                    status_mapping.append ({
+                        'id' : list['id'],
+                        'status' : "Completed"
+                    })
+                else:
+                    pass
+                    #do nothing
+        else:
+            return render_template("404.html", error="problem connecting to Trello API endpoint")           
+    else:
+        return render_template("404.html", error="Trello board 'ToDo' not found!") 
+
+def get_listId(status):
+    for item in status_mapping:
+        if item['status'] == status:
+            listid = item['id']
+    return listid
+
+def get_statusName(id):
+    for item in status_mapping:
+        if item['id'] == id:
+            listStatus = item['status']
+    return listStatus
+
+build_status_mapping()
+
+# error handling for 404
 @app.errorhandler(404)
 def not_found(e):
-    return render_template("404.html")
+    return render_template("404.html", error='resource not found!')
 
-#default
+@app.route('/contact')
+def contact():
+    return render_template('contact.html')
+
+# default
 @app.route('/', methods=['GET'])
 def get_index():
-    all_tasks = get_items()
-    return render_template('index.html', tasks=all_tasks)
-
-@app.route('/', methods=['POST'])
-def post_index():
-    task_title = request.form['title']
-    task_status = request.form['status']
-    add_item(title=task_title, status=task_status)
-    return redirect('/')
+    cards = []
+    for item in status_mapping:
+        list_cards = TrelloCard()
+        result = list_cards.get_List(id = item['id']) 
+        if result.status_code == 200:
+            for card in result.json():
+                if not (card['due'] is None):
+                    formated_date = (card['due']).split("T")[0]
+                else:
+                    formated_date = card['due']
+                cards.append ({
```

Comment:
Rather than setting this to a list of dictionaries, we should use the new Card class that you've created

Reply:
Noted. Thank you. I am now using Card class.